ts/Data/Converters/CSVConverter.ts

```diff
@@ -62,40 +66,139 @@ class CSVParser extends DataParser {
+        options?: CSVConverter.OptionsType
     ) {
         super();
 
-        this.options = merge(CSVParser.defaultOptions, options);
-        this.converter = converter || new DataConverter();
+        this.options = merge(CSVConverter.defaultOptions, options);
     }
 
     /* *
      *
      *  Properties
      *
      * */
+
     private columns: Array<DataTable.Column> = [];
     private headers: Array<string> = [];
     private dataTypes: Array<Array<string>> = [];
     private guessedItemDelimiter?: string;
     private guessedDecimalPoint?: string;
-    private options: CSVParser.ClassJSONOptions;
-    public converter: DataConverter;
 
+    /**
+     * Options for the DataConverter.
+     */
+    public readonly options: CSVConverter.Options;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    /**
+     * Creates a CSV string from the datatable on the store instance.
+     *
+     * @param {CSVConverter.Options} [options]
+     * The options used for the export.
+     *
+     * @return {string}
+     * A CSV string from the table.
+     */
+    public export(
+        store: DataStore,
+        options: CSVConverter.Options = this.options
+    ): string {
+        const { useLocalDecimalPoint, lineDelimiter } = options,
+            exportNames = (this.options.firstRowAsNames !== false);
+
+        let { decimalPoint, itemDelimiter } = options;
+
+        if (!decimalPoint) {
+            decimalPoint = itemDelimiter !== ',' && useLocalDecimalPoint ?
+                (1.1).toLocaleString()[1] :
+                '.';
+        }
+
+        if (!itemDelimiter) {
+            itemDelimiter = decimalPoint === ',' ? ';' : ',';
+        }
+
+        const columns = store.getSortedColumns(options.usePresentationOrder),
+            columnNames = Object.keys(columns),
+            csvRows: Array<string> = [],
+            columnsCount = columnNames.length;
+
+        const rowArray: Array<DataTable.Row> = [];
+
+        // Add the names as the first row if they should be exported
+        if (exportNames) {
+            csvRows.push(columnNames.map(
+                (columnName): string => `"${columnName}"`
+            ).join(itemDelimiter));
+        }
+
+        for (let columnIndex = 0; columnIndex < columnsCount; columnIndex++) {
+            const columnName = columnNames[columnIndex],
+                column = columns[columnName],
+                columnLength = column.length;
+
+            const columnMeta = store.whatIs(columnName);
+            let columnDataType;
+
+            if (columnMeta) {
+                columnDataType = columnMeta.dataType;
+            }
+
+            for (let rowIndex = 0; rowIndex < columnLength; rowIndex++) {
+                let cellValue = column[rowIndex];
+
+                if (!rowArray[rowIndex]) {
+                    rowArray[rowIndex] = [];
+                }
+
+                // Prefer datatype from metadata
+                if (columnDataType === 'string') {
```

Comment:
How about this: 
`if (columnDataType === 'string' || typeof cellValue === 'string') { `

Without the last `else if`?

Reply:
Yes, the code can be improved, but not in this PR. I just copied it over and are glad that it finally works like before. 😄 If you like, you can go over the Converters and Stores and improve it after merge.